core/embed/util/image/inc/util/boot_header.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <trezor_types.h>
+
+// Magic number at the start of the boot header
+#define BOOT_HEADER_MAGIC_TRZQ 0x515A5254  // TRZQ
+// Reserved space for boot header
+#define BOOT_HEADER_MAXSIZE (2 * 8192)
+/** Length of PQ signature in bytes */
+#define BOOT_HEADER_PQ_SIGNATURE_LEN (7856)
+/** Length of EC signature in bytes */
+#define BOOT_HEADER_EC_SIGNATURE_LEN (64)
+/** Number of reserved slots for Merkle proof */
+#define BOOT_HEADER_MERKLE_PROOF_MAXLEN (256)
+/** Maximum accepted size of the bootloader code in bytes */
+#define BOOT_HEADER_CODE_MAXSIZE (1024 * 1024)
+
+/**
+ * SHA-256 fingerprint of the boot header
+ */
+typedef struct {
+  uint8_t bytes[32];
+} boot_header_fingerprint_t;
+
+/**
+ * 4-byte version structure used in the boot header
+ */
+typedef struct __attribute__((packed)) {
+  uint8_t major;
+  uint8_t minor;
+  uint8_t patch;
+  uint8_t build;
+} boot_header_version_t;
+
+/**
+ * Authenticated part of the boot header
+ */
+typedef struct __attribute__((packed)) {
+  /** Magic constant 'TRZQ' */
+  uint32_t magic;
+  /** Hardware model, e.g. 'T3W1'. */
+  uint32_t hw_model;
+  /** Hardware revision, e.g. 1 */
+  uint32_t hw_revision;
+  /** Bootloader version */
+  boot_header_version_t version;
+  /** Bootloader fix version */
+  boot_header_version_t fix_version;
+  /** Minimum previous version taht the device can be updated from */
+  boot_header_version_t min_prev_version;
+  /** Bootloader monotonic version */
```

Comment:
Let's use this place to better explain the meanings of fix version, etc. Also, lets emphasize that `monotonic_version` is not an actual version, but an integer.
```suggestion
  /** Minimum version that the device can be downgraded to without erasing storage. */
  boot_header_version_t fix_version;
  /** Minimum previous version that the device can be updated from when installing this header. */
  boot_header_version_t min_prev_version;
  /** An integer which must not decrease between updates. */
```